<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>findObjectWidget_q</class>
 <widget class="QWidget" name="findObjectWidget_q">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>855</width>
    <height>200</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form1</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_3">
   <item row="1" column="2">
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="0" column="0">
      <widget class="QPushButton" name="findButton">
       <property name="text">
        <string>Find Next</string>
       </property>
       <property name="autoDefault">
        <bool>true</bool>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QPushButton" name="replaceButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Replace</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QPushButton" name="replaceAllButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Replace All</string>
       </property>
      </widget>
     </item>
     <item row="0" column="3">
      <widget class="QPushButton" name="repNextButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Replace &amp;&amp; Find Next</string>
       </property>
      </widget>
     </item>
     <item row="0" column="4">
      <spacer>
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Policy::Expanding</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item row="0" column="2">
    <widget class="QGroupBox" name="buttonGroup2">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Scope for search and replace :</string>
     </property>
     <layout class="QGridLayout">
      <property name="margin" stdset="0">
       <number>4</number>
      </property>
      <item row="0" column="0">
       <widget class="QComboBox" name="srScope">
        <item>
         <property name="text">
          <string>Tree only</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Tree and policy of any firewall</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Policy of any firewall</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Policy of the opened firewall</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="1" rowspan="2">
    <widget class="QGroupBox" name="groupBox5">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string> Replace object</string>
     </property>
     <layout class="QGridLayout">
      <item row="0" column="0">
       <widget class="FWObjectDropArea" name="replaceDropArea">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>80</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>80</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="0" rowspan="2">
    <widget class="QGroupBox" name="groupBox4">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string> Find object</string>
     </property>
     <layout class="QGridLayout">
      <item row="0" column="0">
       <widget class="QComboBox" name="attribute">
        <item>
         <property name="text">
          <string>Name or label</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>IP Address</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>TCP/UDP port</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Protocol number</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>ICMP type</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="0" column="1" rowspan="2">
       <widget class="FWObjectDropArea" name="findDropArea">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>80</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>80</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QComboBox" name="findAttr">
        <property name="editable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="useRegexp">
        <property name="text">
         <string>Use regular expressions</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="2">
    <spacer>
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Policy::MinimumExpanding</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>75</width>
       <height>1</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>FWObjectDropArea</class>
   <extends>QFrame</extends>
   <header>firewallfabrik.gui.drop_area</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>attribute</tabstop>
  <tabstop>findAttr</tabstop>
  <tabstop>srScope</tabstop>
  <tabstop>findButton</tabstop>
  <tabstop>replaceButton</tabstop>
  <tabstop>replaceAllButton</tabstop>
  <tabstop>repNextButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>findButton</sender>
   <signal>clicked()</signal>
   <receiver>findObjectWidget_q</receiver>
   <slot>find()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>findAttr</sender>
   <signal>activated(int)</signal>
   <receiver>findObjectWidget_q</receiver>
   <slot>findAttrChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>findAttr</sender>
   <signal>editTextChanged(QString)</signal>
   <receiver>findObjectWidget_q</receiver>
   <slot>findAttrChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>attribute</sender>
   <signal>activated(int)</signal>
   <receiver>findObjectWidget_q</receiver>
   <slot>findAttrChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>srScope</sender>
   <signal>activated(int)</signal>
   <receiver>findObjectWidget_q</receiver>
   <slot>reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>srScope</sender>
   <signal>activated(int)</signal>
   <receiver>findObjectWidget_q</receiver>
   <slot>scopeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>findDropArea</sender>
   <signal>objectInserted()</signal>
   <receiver>findObjectWidget_q</receiver>
   <slot>objectInserted()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>findDropArea</sender>
   <signal>objectDeleted()</signal>
   <receiver>findObjectWidget_q</receiver>
   <slot>objectDeleted()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>replaceButton</sender>
   <signal>clicked()</signal>
   <receiver>findObjectWidget_q</receiver>
   <slot>replace()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>replaceAllButton</sender>
   <signal>clicked()</signal>
   <receiver>findObjectWidget_q</receiver>
   <slot>replaceAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>repNextButton</sender>
   <signal>clicked()</signal>
   <receiver>findObjectWidget_q</receiver>
   <slot>replaceNext()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>replaceDropArea</sender>
   <signal>objectDeleted()</signal>
   <receiver>findObjectWidget_q</receiver>
   <slot>replaceDisable()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>replaceDropArea</sender>
   <signal>objectInserted()</signal>
   <receiver>findObjectWidget_q</receiver>
   <slot>replaceEnable()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>find()</slot>
  <slot>findAttrChanged(int)</slot>
  <slot>findAttrChanged(QString)</slot>
  <slot>reset()</slot>
  <slot>scopeChanged()</slot>
  <slot>objectInserted()</slot>
  <slot>objectDeleted()</slot>
  <slot>replace()</slot>
  <slot>replaceAll()</slot>
  <slot>replaceNext()</slot>
  <slot>replaceDisable()</slot>
  <slot>replaceEnable()</slot>
 </slots>
</ui>
